src/sql_generator.cpp

```diff
@@ -230,14 +230,30 @@ void alter_table(duckdb::Connection &con, const table_def &table,
   }
 }
 
+std::string
+make_full_column_list(const std::vector<const column_def *> &columns_pk,
+                      const std::vector<const column_def *> &columns_regular) {
+  std::ostringstream full_column_list;
+  if (!columns_pk.empty()) {
+    write_joined(full_column_list, columns_pk, print_column);
```

Comment:
What does [`write_joined`](https://github.com/MotherDuck-Open-Source/motherduck-fivetran-connector/blob/c7dac317910220b55df45bd879a4ec6916a8e9ec/src/sql_generator.cpp#L22) do? As far as I can tell it prints the column names separated by `",  "` to the `full_column_list` stream; why are we adding the comma + spaces again on line 240? And why is it called again on line 242? Sorry if I'm not reading this correctly (the indentation is a bit confusing to read on GitHub)

Reply:
`write_joined` takes a bunch of column definitions and writes a list of (maybe transformed) column names with comma separators in the middle.

In this case, I don't have one list of columns -- I have two separate lists (well, vectors) of them, one with only primary key columns, and the other -- with the rest of columns. So I could make a big vector combining the two, and then call `write_joined()` once, but instead I am calling `write_joined` on just the primary key columns, adding a special comma-space-space separator*

So what happens is, in a table with  "id",  "name" and "description" columns where only "id" is a primary key, line 238 will only output `"id"` (without any commas since it's only one value), line 240 will output `,  ` , and line 242 will output `"name", "description"`.

Since DuckDB is very permissive about trailing commas, I could probably change `write_joined()` to always output a trailing comma, making printing the extra `,  ` unnecessary.

* that part is not needed. I added it because I needed a comma-space separator between two sets of columns anyway, and adding a "special" two space separator will give us some way of troubleshooting if no primary columns came though. We only have backend logging for SQL queries, so I added a small hidden troubleshooting space.

Reply:
> could probably change write_joined() to always output a trailing comma
That sounds good to me!